src/coreclr/jit/lowerxarch.cpp

```diff
@@ -42,61 +42,29 @@ void Lowering::LowerRotate(GenTree* tree)
 //    This involves:
 //    - Handling of contained immediates.
 //    - Widening operations of unsigneds.
-
+//
 void Lowering::LowerStoreLoc(GenTreeLclVarCommon* storeLoc)
 {
-    // Try to widen the ops if they are going into a local var.
-    if ((storeLoc->gtOper == GT_STORE_LCL_VAR) && (storeLoc->gtOp1->gtOper == GT_CNS_INT))
+    // Most small locals (the exception is dependently promoted fields) have 4 byte wide stack slots, so
+    // we can widen the store, if profitable. This optimization is not relevant for register candidates.
```

Comment:
ARM has the same "optimization" (which does not make much sense on ARM I think). Should it be adjusted? It does not have the `lvDoNotEnregister` heuristic check, but it probably makes sense to remove it entirely.

Reply:
> ARM

I've investigated this question a while back.

1) For ARM64 this widening is useless. I'll `#ifdef` it out and delete the LA64 copy.
2) For ARM, it can be useful because 4-byte-wide stores can be encoded in 2 bytes in more cases than the small ones. I'll make the code look more like this new x86/x64 version.

> It does not have the lvDoNotEnregister heuristic check, but it probably makes sense to remove it entirely.

Yes... Not sure why I put that in; it's true that the "optimization" is only relevant for in-memory locals, but even candidates can end up spilled. I'll remove it.